npm/pre-install.js

```diff
@@ -1,10 +1,23 @@
+// @ts-check
+import {familySync, GLIBC, MUSL} from "detect-libc"
+import get_matched_platform from "./utils.js"
+
 const os = process.platform
 const arch = process.arch
+const libcFamily = familySync()
+
+let libc = ""
+if (os === "win32") {
+  libc = "msvc"
+} else {
+  libc = libcFamily === GLIBC ? "gnu" : libcFamily === MUSL ? "musl" : ""
+}
+
+const matched_platform = get_matched_platform(os, arch, libc)
 
-const dependency = Object.keys(optionalDependencies).find((name) => name.includes(`${os}-${arch}`))
-if (!dependency) {
+if (matched_platform == null) {
   const redColor = "\x1b[31m"
   const resetColor = "\x1b[0m"
-  console.error(`${redColor} Tailcall does not support platform ${os} arch ${arch} ${resetColor}`)
+  console.error(`${redColor} Tailcall does not support platform - ${os}, arch - ${arch}, libc - ${libc} ${resetColor}`)
```

Comment:
Why are we making changes in this file?

Reply:
just to be verbosea and also
some times libc difference also makes platform not supportable.